napari/_vispy/vendored/volume.py

```diff
@@ -375,11 +475,42 @@
         """,
 )
 
+# This is an iso shader for categorical data (e.g., label images)
+ISO_CATEGORICAL_SNIPPETS = dict(
+    before_loop="""
+        vec4 color3 = vec4(0.0);  // final color
+        vec3 dstep = 1.5 / u_shape;  // step to sample derivative
+        gl_FragColor = vec4(0.0);
+    """,
+    in_loop="""
+        // background is assumed to be 0
+        if (val > 0) {
```

Comment:
This works for labels in napari but is not a safe default generally, we should probably expose it as a variable on the `VolumeVisual` 
I know you already played with a 'background_colour` shading filter, maybe as a separate PR we should expose a `background_value` on the `VolumeVisual` which is ignored for all rendering modes

Reply:
Good point. I can add the background_value as a uniform for now and expose it on the VolumeVisual.

The one tricky thing about supporting non-zero background values is that we then need to compare if the `val != background_value`. I think this will require us to also add a tolerance for that comparison since `val` is a float. That is something like: `if abs(val - background_value) < tolerance`. Not a huge deal, but we do need to add another parameter.

Reply:
We now use `u_threshold` (same uniform for setting the iso level in the standard iso rendering) to set the background level. I introduced a couple of functions to check equality of floats with a threshold (0.0001). That threshold is currently a uniform with a pre-defined value. We can expose this on the visual if we desire. Thoughts, @alisterburt ?